samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/main.c

```diff
@@ -110,8 +109,8 @@ static void light_default_status_init(void)
 
 	default_tt = gen_def_trans_time_srv_user_data.tt;
 
-	target_lightness = lightness;
-	target_temperature = temperature;
+	init_lightness_target_values();
+	init_temp_target_values();
```

Comment:
Why do you do the initialisation at runtime instead of build time? It seems all this code is called from main() before loading settings, i.e. you always know the state up-front, i.e. there's probably a lot of redundant code in these init functions (in the branches that will never be taken)

Reply:
No. Please refer main.c, I have called light_default_status_init() after loading settings.

Reply:
I don't see that. I see that you call bt_enable() asyncrhously, passing it the bt_ready callback. You then proceed with calling light_default_status_init() right after that, i.e. without any guarantee that bt_ready (which calls settings_load) will have been called.

Reply:
As of now, let it be. I will take care of it in next PR (Since it is going on from very long time).
If you have quick suggestions then please suggest it so that I can quickly repair the things.

Reply:
You could call bt_enable synchronously by passing NULL to it, and then explicitly call bt_ready from main() once bt_enabled has returned success.

Reply:
Thanks for the suggestion. I've added 6th commit as per it.